protocol/test-serialization/src/fuzz.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+#[cfg(test)]
+mod tests {
+    use crate::fuzz_test;
+    use bin_prot::Deserializer;
+    use mina_rs_base::types::*;
+    use rand::prelude::*;
+    use serde::Deserialize;
+    use wasm_bindgen_test::*;
+
+    #[test]
+    #[wasm_bindgen_test]
+    fn test_corrupted_deserialization() {
+        fuzz_test!(
+            ExternalTransition
+
+            ProtocolState
+            ProtocolStateBody
+            BlockchainState
+            ConsensusState
+            ProtocolConstants
+
+            ProtocolStateProof
+            ProofStatement
+            PrevEvals
+            Proof
+            ProofMessages
+            ProofOpenings
+
+            StagedLedgerDiff
+            StagedLedgerDiffTuple
+            StagedLedgerPreDiffTwo
+
+            DeltaTransitionChainProof
+
+            Option<ProtocolVersion>
+            ()
+        );
+    }
+
+    #[macro_export]
+    macro_rules! fuzz_test {
+        ($($ty: ty) *) => {
+            $(
+                let mut rng = rand::thread_rng();
```

Comment:
Each time this runs it will produce different random bytes.  Therefore, if it fails and you try to fix and rerun it, then the you will not get the same test case.  It's usually better to be deterministic, i.e., seed the Rng so that it produces the same random bytes each time it's run.   For example like this `let mut rng = StdRng::from_seed([0; 32]);`

Reply:
Fixed

Reply:
Nice - thanks